user_signup/main.py

```diff
@@ -0,0 +1,86 @@
+import webapp2
+import jinja2
+import os
+import re
+
+jinja_env = jinja2.Environment(
+	loader=jinja2.FileSystemLoader(os.path.dirname(__file__)), 
+	autoescape=True)
+
+USER_RE = re.compile(r"^[a-zA-Z0-9_-]{3,20}$")
+PW_RE = re.compile(r"^.{3,20}$")
+EMAIL_RE = re.compile(r"^[\S]+@[\S]+.[\S]+$")
```

Comment:
Didn't write them - they were given so we could focus on the other parts of this. I'll use better ones in the final blog project.